doc/development/plugins.rst

```diff
@@ -232,6 +232,91 @@ or can include in-built transforms such as:
+>>> batch[0].circuit
+[CNOT(wires=[0, 1]), expval(Z(0))]
+
+However, if the device native gate set is unreachable with the default decompositions defined in PennyLane,
+an error will be raised. In this case, you may need to override the decompositions of certain operators
+via the ``decomposer`` argument. 
+
+For example, consider we have a device with ``RX``, ``RY`` and ``IsingXX`` as native gates but we want
+to execute a circuit written in terms of ``CNOT`` s. Then, we can define a decomposition for ``CNOT`` 
+(e.g., ``custom_decomposer``) and pass it to the decomposer:
+
+.. code-block:: python
+
+    def stopping_condition(op):
+        return op.name in {"IsingXX", "RX", "RY"}
+
+    def custom_decomposer(op):
+        if isinstance(op, qml.CNOT):
+            wires = op.wires
+            return [
+                qml.RY(np.pi/2, wires=wires[0]),
+                qml.IsingXX(np.pi/2, wires=wires),
+                qml.RX(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[1])
+            ]
+        return op.decomposition()
+
+>>> circuit = qml.tape.QuantumScript([qml.CNOT(wires=(0,1))], [qml.expval(qml.Z(0))])
+>>> batch, fn = decompose(circuit, stopping_condition, decomposer=custom_decomposer)
+>>> batch[0].circuit
+[RY(1.5707963267948966, wires=[0]),
+IsingXX(1.5707963267948966, wires=[0, 1]),
+RX(-1.5707963267948966, wires=[0]),
+RY(-1.5707963267948966, wires=[0]),
+RY(-1.5707963267948966, wires=[1]),
+expval(Z(0))]
+
+An alternative approach for overriding the decompositions of certain operators is by taking advantage of
+the new graph-based decomposition algorithm.
+In this case the ``target_gates`` kwarg needs to be specified in the :func:`~pennylane.devices.preprocess.decompose`
+transform. Note that the stopping condition function defines whether an operator should be decomposed,
+while the ``target_gates`` defines the set of operator types that the graph-based decomposition
+algorithm needs to target.
+
+With this approach, the decomposition for the CNOT needs to be specified as a quantum function, ``decompose_cnot``, and
+registered with ``qml.add_decomps``:
+
+.. code-block:: python
+
+    qml.decomposition.enable_graph()
+
+    def stopping_condition(op):
+    return op.name in {"IsingXX", "RX", "RY"}
+
+    target_gates={qml.IsingXX, "RX", "RY"}
+
+    @qml.register_resources({qml.RY: 3, qml.RX: 1, qml.IsingXX: 1})
+    def decompose_cnot(wires, **_):
+        qml.RY(np.pi/2, wires=wires[0]),
+        qml.IsingXX(np.pi/2, wires=wires),
+        qml.RX(-np.pi/2, wires=wires[0]),
+        qml.RY(-np.pi/2, wires=wires[0]),
+        qml.RY(-np.pi/2, wires=wires[1])
+
+    qml.add_decomps(qml.CNOT, decompose_cnot)
+
+>>> circuit = qml.tape.QuantumScript([qml.CNOT(wires=(0,1))], [qml.expval(qml.Z(0))])
+>>> batch, fn = decompose(circuit, stopping_condition, device_wires=[2], target_gates=target_gates)
+>>> batch[0].circuit
```

Comment:
Same comment here. This entire section is about setting up the transform program so our examples should be around that.

Reply:
Fair point. I'll fix the examples 👍 
I ended up modifying them with the idea that they could be ran, but I agree that this looks more like documentation for [qml.devices.preprocess.decompose](https://docs.pennylane.ai/en/stable/code/api/pennylane.devices.preprocess.decompose.html)

Reply:
Made changes here https://github.com/PennyLaneAI/pennylane/pull/8492/commits/b5d10c938bf91aa7efb2329da0f3e99c10304998